coding/sha1.cpp

```diff
@@ -1,20 +1,36 @@
 #include "coding/sha1.hpp"
+#include "coding/base64.hpp"
 
 #include "coding/internal/file_data.hpp"
 #include "coding/reader.hpp"
 
 #include "base/assert.hpp"
 #include "base/logging.hpp"
-#include "base/macros.hpp"
 
-#include "3party/liboauthcpp/src/SHA1.h"
-#include "3party/liboauthcpp/src/base64.h"
+#include <boost/core/bit.hpp>
+#include <boost/uuid/detail/sha1.hpp>
 
 #include <algorithm>
 #include <vector>
 
 namespace coding
 {
+namespace
+{
+SHA1::Hash ExtractHash(boost::uuids::detail::sha1 & sha1)
+{
+  uint32_t digest[5];
+  sha1.get_digest(digest);
+  for (auto & b : digest)
+    b = boost::core::byteswap(b);
```

Comment:
1. Why do you think a byte swap is necessary compared to the previous implementation? Why are implementations different?
2. How critical is ExtractHash performance? Does it make sense to paste here boost's implementation and fix it to avoid bits rotation? Or maybe paste here the necessary part from liboauth implementation? Which implementation is faster, or are they the same?

Reply:
> Why do you think a byte swap is necessary compared to the previous implementation?

Digest from `liboauthcpp` is little-endian (controlled by `SHA1_LITTLE_ENDIAN`).
https://github.com/organicmaps/organicmaps/blob/967ffc1384617d31ab6d4d340bbfb2c7909fb2ce/3party/liboauthcpp/src/SHA1.h#L55-L57
Digest from `boost` is little-endian.

> Why are implementations different?

I can only speculate but it's probably just a preference.

> How critical is ExtractHash performance?

Not critical. It's used to verify downloaded map data in a separate thread.

> Does it make sense to paste here boost's implementation and fix it to avoid bits rotation? Or maybe paste here the necessary part from liboauth implementation?

It's up to you to decide ;)

> Which implementation is faster, or are they the same?

Since the code is not critical, I don't see a reason to write benchmarks.

Reply:
> Digest from boost is little-endian.

You likely meant big-endian? Does the result depend on the current arch?

Maybe use boost::endian::endian_reverse_inplace ?

Reply:
> You likely meant big-endian?

Yes.

> Does the result depend on the current arch?

Hmm I have no idea how to verify that.